design/specs/visual-query-builder.md

```diff
@@ -0,0 +1,236 @@
+
+### Changing the base table
+
+The user should choose a base table at the start of the query building process. Changing it after the input columns have been selected will result in the loss of all progress.
+
+A warning should be issued to users so that they are aware of the implications of making this change.
+
+### 1.1 Base Table Options
+
+The base table selector will show a list of all the tables that are currently available in the schema. (At first, this list will not display views. View exploration will be added in later iterations.)
+
+To choose a table, the user must first locate it on the list and then click on the desired table option.
+
+When you select the base table, the interface controls for additional steps (e.g., column selection, filters) that were previously disabled will become available. Before selecting a base table, no action is available.
+
+After making a selection, the user will be prompted to continue with the input column configuration, starting with column selection.
+
+---
+Wireframes
+
+[Selecting a Base Table](https://share.balsamiq.com/c/esUftomoFsZiRhZDMYdoPP.png)
+
+[Base Table Selected](https://share.balsamiq.com/c/me5CfGGeX3R35x2otLqbpu.png)
+
+## 2. Creating the Input Table
+
+The input table is built by selecting columns from the base table or by generating new ones using formulas.
+
+Based on the order of selection, these columns are added to the result table. Columns can be renamed by users as they see fit.
+
+The system will name columns by default to indicate their source table and column using the [naming convention](#column-naming-convention).
+
+Some options, such as filters and aggregations, may be available for columns that link to multiple records to help users define which data they want to show and how. The system will automatically detect the type of column and present users with the available options.
+
+### 2.1. Adding Input Columns
+
+To begin adding columns to a query configuration panel, the user must first select the `Add Column` button situated in the `Columns` portion of the query configuration panel. The `Column Selector` component will be displayed in the inspector panel as a result of this action.
+
+The `Column Selector` will display a list of all available columns as well as those already in use, if available. In addition, a complete list of all formulas will be available.
+
+The columns in `Column Selector` are listed in a hierarchical structure based on the links that exist in both directions between the base table and other tables. The icons indicate whether the linked records are a single or multiple record collection.
+
+Links are presented as expandable sections containing the columns from the linked tables.
+
+When adding input columns, the user can add them using the list controls or drop the columns directly into the result table.
+
+---
+Wireframes
+
+[Adding Columns](https://share.balsamiq.com/c/oyxTXxqSh8rLqU3JY71DWY.png)
```

Comment:
Yes, they can add a column again. They may, however, change the column names and even the data types of the columns that have been selected (through aggregations or summarization). The 'In use' section would allow users to quickly see which columns are being used in the view. Consider a user inspecting an existing view that they did not create and simply wanting to list all of the sources.